src/mono/mono/mini/debugger-agent.c

```diff
@@ -6847,6 +6870,16 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
 	case CMD_VM_STOP_BUFFERING:
 		/* Handled in the main loop */
 		break;
+	case MDBGPROT_CMD_VM_READ_MEMORY: {
+		guint8* memory = (guint8*) decode_long (p, &p, end);
+		int size = decode_int (p, &p, end);
+		buffer_add_byte_array (buf, memory, size);
```

Comment:
Does this need segfault protection like in https://github.com/mono/mono/issues/15612

Reply:
If final user tries to evaluate an arbitraty address we would need to protect, but I wouldn't like to use this because icordebug uses this(ReadMemory) a lot, and create a file to check memory address would be really slow... I'm planning to study how coreclr checks if it is a valid memory. But I was not planning to do it in this PR, do you think I should add the segfault protection for now and then remove when I have a better solution?